### DISCUSSION TITLE
running radiopadre on a Kubernetes cluster

### DISCUSSION THREAD

**Discussion by o-smirnov** [ratt-ru/radiopadre-client#54] [Show and tell]
OK @landmanbester you'll like this especially. Check out radiopadre-client branch [b1.2.2](https://github.com/ratt-ru/radiopadre-client/tree/b1.2.2), now with Kubernetes support. All you need is the client branch, and pip install kubernetes in the virtualenv.

Then run like this:

```
run-radiopadre -K rarg-test-compute-efs-pvc:parrot --k8s-node-selector rarg/node-class=compute,rarg/instance-type=m5.4xlarge -e
```

``-K`` selects the K8s backend. ``rarg-test-compute-efs-pvc`` refers to the PVC to connect to. ``parrot`` refers to a subdirectory on this PVC. This particular ``--k8s-node-selector`` setting specifies a non-default instance type to schedule the pod on -- I have a feeling an ``m5.4xlarge`` is probably needed to run CARTA smoothly (I should probably look into specifying this in terms of CPU and RAM requests). Finally, ``-e`` will save your given command line settings to your ``~/.radiopadre/radiopadre-client.config``, so that next time you can run simply

```
run-radiopadre -K rarg-test-compute-efs-pvc:parrot
```

...without specifying any other command-line options.

Other useful options may be ``--k8s-context`` and ``--k8s-namespace``. See also ``--help`` for others.

Feedback solicited.

**Comment by o-smirnov** [comment]
OK, added this: you can specify ``--k8s-cpu-request 2 --k8s-ram-request 8Mi``, rather than worrying about node selectors.

Might need to adjust the RAM values upwards if CARTA is struggling.

**Comment by JSKenyon** [comment]
I have a possibly very stupid question - did you check in the source for `radiopadre_client.kube`? I ran into:
```
Traceback (most recent call last):
  File "/home/jonathan/venvs/awsrpenv/bin/run-radiopadre", line 404, in <module>
    import radiopadre_client.kube
ModuleNotFoundError: No module named 'radiopadre_client.kube'
```
and cannot find the import in question.

**Reply by JSKenyon** [reply]
@o-smirnov If you have a moment, could you please let me know if I have missed something obvious?

**Comment by o-smirnov** [comment]
@JSKenyon done :facepalm:. Sorry about the delay.

**Reply by JSKenyon** [reply]
Wait - should I be using b1.2.1 or b1.2.2? I see a commit in b1.2.1 but have been using b1.2.2!

**Comment by JSKenyon** [comment]
Hurray! This seems to be working for me now! Should make experimenting/wreckage analysis simpler.

**Comment by landmanbester** [comment]
With new account run
```
run-radiopadre -K admin-efs-pvc: --k8s-node-selector rarg/node-class=compute,rarg/instance-type=m5.large -e
```
